src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Engine.Incrementals
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Exceptions;
+    using Microsoft.DocAsCode.Utility;
+
+    public sealed class ChangeList
+    {
+        private readonly List<ChangeItem> _list = new List<ChangeItem>();
+
+        public static ChangeList Parse(string tsvFile, string baseDir)
+        {
+            if (tsvFile == null)
+            {
+                throw new ArgumentNullException(nameof(tsvFile));
+            }
+            if (baseDir == null)
+            {
+                throw new ArgumentNullException(nameof(baseDir));
+            }
+            if (File.Exists(tsvFile))
+            {
+                throw new FileNotFoundException("File not found.", tsvFile);
+            }
+            return ParseCore(tsvFile, baseDir);
+        }
+
+        public void Add(string filePath, ChangeKind kind)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+            if (filePath.Length == 0)
```

Comment:
exception is different.

---
In reply to: [73995730](https://github.com/dotnet/docfx/pull/586#discussion_r73995730) [](ancestors = 73995730)